funnel/collection/main.py

```diff
@@ -0,0 +1,97 @@
+# Lets you tokenize a lazy object
+import typing
+import warnings
+
+import intake
+import pydantic
+from dask.base import tokenize
+
+from ..config import config as default_config
+from ..metadata_db.main import MemoryMetadataStore, SQLMetadataStore
+
+
+class OriginDict(pydantic.BaseModel):
+    collection_name: str
+    esm_collection_json: str
+    esm_collection_query: dict
+    operators: list
+    operator_kwargs: list
+    esm_collection_key: str = None
+    variable: str = None
+
+
+@pydantic.dataclasses.dataclass
+class Collection:
+    # rename metadatastore - cache_database
+    collection_name: str
+    esm_collection_json: str
+    esm_collection_query: dict
+    operators: list = None
+    operator_kwargs: list = None
+    serializer: str = 'xarray.zarr'
+    metadata_store: typing.Union[MemoryMetadataStore, SQLMetadataStore] = None
+    kwargs: dict = None
+
+    def __post_init_post_parse__(self):
+        self.operators = self.operators or []
+        self.operator_kwargs = self.operator_kwargs or [{}]
+        self.metadata_store = default_config.metadata_store or self.metadata_store
+        self.origins_dict = OriginDict(
+            collection_name=self.collection_name,
+            esm_collection_json=self.esm_collection_json,
+            esm_collection_query=self.esm_collection_query,
+            operators=self.operators,
+            operator_kwargs=self.operator_kwargs,
+        ).dict()
+        self.catalog = intake.open_esm_datastore(self.esm_collection_json)
+        self.variable_column_name = self.catalog.col.aggregation_info.variable_column_name
+        self.base_variables = set(self.catalog.df.variable)
+        self.variable = self.esm_collection_query.get(self.variable_column_name, None)
+
+    def to_dataset_dict(self, variable, compute=False):
+        """Returns a dictionary of datasets similar to intake-esm"""
+        dsets = {}
+        if not isinstance(variable, list):
+            variable = [variable]
+
```

Comment:
Is the idea here to turn a string into a list with one item? If so, could we just check if it's a string instead

```python
if isinstance(variable, str):
```